lib/updater.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class Updater
+  def initialize(user_interface, database)
+    @user_interface = user_interface
+    @database = database
+  end
+
+  attr_reader :user_interface, :database
+
+  def run
+    return user_interface.display_no_contacts_message, user_interface.continue if database.database_empty?
```

Comment:
You won't see two instructions separated by comma in a return followed by a conditional a lot.

and that is because of the cognitive complexity it takes for a human brain to parse

so sometimes it's better to be more verbose

```ruby
if database.database_empty?
  user_interface.display_no_contacts_message, 
  user_interface.continue 
  return 
end
```
you could also check with the other classes if you could add `continue` inside of `display_no_contacts_message`

and you can also put the loop in its own method

Reply:
I did have a look at the display_no_contacts_message usage and I would need to create a separate method if the solution is to combine them. I'm a little confused on where that behaver is best addressed, it does affect the display but is it a userinterface concern?

Reply:
at the moment the finder and the pager are the ones using the method `display_no_contacts_message`. 

So the question would be, to check if we could move `continue` inside of `display_no_contacts_message` and still use the finder and pager in the same way, i.e. would the logic of the finder and the pager had to be updated for that to work.

No new methods would be added:

```ruby
  def display_no_contacts_message
    output.puts Constants::NO_CONTACTS_MESSAGE
    continue
  end
```

instead of 

```ruby
  def display_no_contacts_message
    output.puts Constants::NO_CONTACTS_MESSAGE
  end
```